### PR TITLE
[docs] Adjust markdown docs generator to move files to dedicated directory

### PR DESCRIPTION
**Pull request by lucgrabowski**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Adjust `markdown_docs_generator` to move files to `docs/generated` directory in docs repository

**This PR should be considered together with:** https://github.com/fastlane/docs/pull/1095

### Description
Adjust `update_docs` lane in `Fastfile` and `generate!` in `markdown_docs_generator` to move auto-generated files to `docs/generated` directory in docs repository

Urls in actions documentation will not change. There is a PR in docs repository to copy files from generated to correct locations

### Testing Steps

Update `Gemfile` and run `bundle install`:
```ruby
gem "fastlane", :git => "https://github.com/lucgrabowski/fastlane.git", :branch => "lucgrabowski-move-autogenerated-files-to-dedicated-directory"
```

### CODE REVIEW

File: fastlane/lib/fastlane/documentation/markdown_docs_generator.rb [hunk 1]
```diff
@@ -134,15 +134,16 @@ def generate!(target_path: nil)
       require 'yaml'
       FileUtils.mkdir_p(target_path)
       docs_dir = File.join(target_path, "docs")
+      generated_actions_dir = File.join("generated", "actions")
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/documentation/markdown_docs_generator.rb:137]: Does this directory need a `FileUtils.mkdir_p` on it? 🤔

**Review by joshdholtz — CHANGES_REQUESTED** [review]
The validate docs CI job is failing right now and I _think_ it might be because the `generated` and `generated/actions` directories don't exist yet? It seems like a `mkdir_p` is needed but unsure if that is the actual fix 🤷‍♂️ 

Thoughts? 😇

File: fastlane/lib/fastlane/documentation/markdown_docs_generator.rb (hunk 1 above)
Comment (lucgrabowski) [reply fastlane/lib/fastlane/documentation/markdown_docs_generator.rb:137]: I think that you're right, it wouldn't be needed on related docs branch.

**Comment by lucgrabowski** [comment]
@joshdholtz Should "Customize branch of docs repository" commit be reverted? It's not needed.
And thanks for help 🚀

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Comment by lucgrabowski** [comment]
@joshdholtz Thanks for help with this PR 💪 and fixing `mkdocs` config :)

**Review by joshdholtz — APPROVED** [review]
Whoops! I thought I merged this but I guess not 😱 I was moving to a new house all last week so wasn't able to do much work. Merging this now and thanks again for making this change!

**Pull request merged**